src/proxy.rs

```diff
@@ -148,54 +150,93 @@ pub struct Proxy {
 #[derive(Clone)]
 pub struct ScopedSecretManager {
     cert_manager: Arc<SecretManager>,
-    allowed: Option<Arc<WorkloadInfo>>,
+    allowed: Arc<WorkloadInfo>,
```

Comment:
(probably out of context thought) ScopedSecretManager could probably be removed or mostly mooted if `allowed` were just an Identity instead of a WorkloadInfo.

This would require waiting to create the proxy until we got both the XDS push and the WorkloadInfo, so we could construct and propagate the Identity tho.